cmd/ipfs/Rules.mk

```diff
@@ -12,7 +12,10 @@ PATH := $(realpath $(d)):$(PATH)
 # DEPS_OO_$(d) += merkledag/pb/merkledag.pb.go namesys/pb/namesys.pb.go
 # DEPS_OO_$(d) += pin/internal/pb/header.pb.go unixfs/pb/unixfs.pb.go
 
-$(d)_flags =-ldflags="-X "github.com/ipfs/go-ipfs/repo/config".CurrentCommit=$(shell git rev-parse --short HEAD)" 
+CONFIG_COMMIT ?= github.com/ipfs/go-ipfs/repo/config.CurrentCommit=$(shell git rev-parse --short HEAD)
+CONFIG_PLUGIN_PATH = github.com/ipfs/go-ipfs/repo/config.SystemPluginPath=${PLUGIN_PATH}
```

Comment:
I would do 3 things here:

1 . Scope the `CONFIG_COMMIT` to local as `$(d)_commit`
2. Scope the `CONFIG_PLUGIN_PATH` to local as `$(d)_plugin_path`
3. Create file: `config.mk` in root of the project that would contain user configurable variables (using `?=` for defaults with description) and phony target that would print those variables.  Variables in this file would be prefixed with `CONFIG_`. This file would be included early in either `Makefile` or `Rules.mk` (probably `Makefile`).

This way global namespace is not randomly polluted which we are trying to avoid.